src/delay.rs

```diff
@@ -38,22 +38,20 @@ impl Delay {
     /// The default timer will be spun up in a helper thread on first use.
     #[inline]
     pub fn new(dur: Duration) -> Delay {
-        Delay::new_at(Instant::now() + dur)
+        Delay::new_handle(Instant::now() + dur, Default::default())
     }
 
-    /// Creates a new future which will fire at the time specified by `at`.
-    ///
-    /// The returned object will be bound to the default timer for this thread.
-    /// The default timer will be spun up in a helper thread on first use.
+    /// Return the `Instant` when this delay will fire.
     #[inline]
-    pub fn new_at(at: Instant) -> Delay {
-        Delay::new_handle(at, Default::default())
+    pub fn when(&self, timeout: &Delay) -> Instant {
+        timeout.when
```

Comment:
this should be `pub fn when(&self) -> Instant {  self.when }`?

Reply:
You're entirely right. Thanks for pointing that out! Filed #38 to address that.